### ISSUE TITLE
Rendering of not changed canvas objects after an event (after eg. Button move over)

### ISSUE THREAD

**Issue by renlite** [fyne-io/fyne#3211]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

After an event occurs all canvas objects (eg. line, rectangle) on the screen will be rerendered, even though only one Widget has changed. This may affect the painting performance especially on Web (GL ES).

### How to reproduce

1. Set a breakpoint in internal/painter/gl/draw.go in the method ```func (p *painter) drawObject(o fyne.CanvasObject, pos fyne.Position, frame fyne.Size) {...}```
2. Move over a Button, click the Button and leave the Button
3. The Button events will rerender all object on the screen, not only the objects of the Button that has changed

### Screenshots

![image](https://user-images.githubusercontent.com/64831420/185378969-ff9fd0d9-5373-4313-9455-192362fc6747.png)

### Example code

```go
func (p *painter) drawObject(o fyne.CanvasObject, pos fyne.Position, frame fyne.Size) {
	switch obj := o.(type) {
	case *canvas.Circle:
		p.drawCircle(obj, pos, frame)
	case *canvas.Line:
		p.drawLine(obj, pos, frame)
	case *canvas.Image:
		p.drawImage(obj, pos, frame)
	case *canvas.Raster:
		p.drawRaster(obj, pos, frame)
	case *canvas.Rectangle:
		p.drawRectangle(obj, pos, frame)
	case *canvas.Text:
		p.drawText(obj, pos, frame)
	case *canvas.LinearGradient:
		p.drawGradient(obj, p.newGlLinearGradientTexture, pos, frame)
	case *canvas.RadialGradient:
		p.drawGradient(obj, p.newGlRadialGradientTexture, pos, frame)
	}
}
```

### Fyne version

branche: develop

### Go compiler version

1.17

### Operating system

Windows

### Operating system version

Windows 10

### Additional Information

-

**Comment by gavinwade12** [comment]
Could you please provide a minimal version of _your_ example code so this issue can easily be reproduced? :)

**Comment by renlite** [comment]
You can find the example in PR #3121: [example/flexrect.go](https://github.com/fyne-io/fyne/pull/3121/files#diff-90431526b53b1b9510c32b3617ac4c5fcb8d8d62de400bf6729cd73bf778d734)
In the draw.go there are ```println()```commands included.
But the issue should happen on the master branche too.

**Comment by andydotxyz** [comment]
Sounds like something is invalidating a texture that it shouldn't...
We don't have partial re-render so when the UI must paint we have to paint every element (from cache).
In the case of a button hover only the rectangle should be rendered, though all of them will be painted.

**Comment by andydotxyz** [comment]
Resolved on develop ready for v2.3.0
